lib/Data/Database.php

```diff
@@ -691,21 +752,35 @@ private static function _getPrimaryKeyClauses($key = 'dataid')
      */
     private static function _getDataType()
     {
-        return self::$_type === 'pgsql' ? 'TEXT' : 'BLOB';
+        return self::$_type === 'pgsql' ? 'TEXT' : (self::$_type === 'oci' ? 'VARCHAR2(4000)' : 'BLOB');
     }
 
     /**
      * get the attachment type, depending on the database driver
      *
-     * PostgreSQL uses a different API for BLOBs then SQL, hence we use TEXT
+     * PostgreSQL and OCI use different APIs for BLOBs then SQL, hence we use TEXT and CLOB
      *
      * @access private
      * @static
      * @return string
      */
     private static function _getAttachmentType()
     {
-        return self::$_type === 'pgsql' ? 'TEXT' : 'MEDIUMBLOB';
+        return self::$_type === 'pgsql' ? 'TEXT' : (self::$_type === 'oci' ? 'CLOB' : 'MEDIUMBLOB');
+    }
+
+    /**
+     * get the meta type, depending on the database driver
+     *
+     * OCI can't even accept TEXT so it has to be VARCHAR2(200)
+     *
+     * @access private
+     * @static
+     * @return string
+     */
+    private static function _getMetaType()
+    {
+        return self::$_type === 'oci' ? 'VARCHAR2(4000)' : 'TEXT';
```

Comment:
Ahhh here is the 4000 chars limit? So why 4000? Can't we use a bigger value? Or something like `VARCHAR2(MAX)` (again I don't know the details of Oracle DBs, but feel free to link it/the actual reason in the PHPDoc e.g.)

Also the PHPDoc is currently wrong, it says it uses `VARCHAR2(200)`…

Reply:
It seems to be the maximum for their VARCHAR2 type, at least by default: https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/datatype-limits.html#GUID-963C79C9-9303-49FE-8F2D-C8AAF04D3095

4000 bytes is pretty generous, the metadata is pretty limited as most of it's contents are removed from it and stored in dedicated columns. What is left is converted to a JSON string.